public/app/api/legacy/user/api.ts

```diff
@@ -0,0 +1,25 @@
+import { createApi } from '@reduxjs/toolkit/query/react';
+
+import { createBaseQuery } from 'app/api/createBaseQuery';
+
+export const legacyUserAPI = createApi({
```

Comment:
Added an RTKQ API even though it'll be gone after app platform, as it made it a lot cleaner to implement the hooks, with switching between the legacy/k8s APIs.

Also fixed an existing issue with DashList where we would call `/api/stars` for each dashboard (and I didn't want to write some custom caching logic). This also makes it cleaner for cache invalidation etc